internal/raft/storage/store.go

```diff
@@ -0,0 +1,412 @@
+package storage
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"github.com/coreos/etcd/pkg/pbutil"
+	"github.com/coreos/etcd/raft"
+	pb "github.com/coreos/etcd/raft/raftpb"
+	"github.com/dgraph-io/badger/v3"
+	bdgr_opts "github.com/dgraph-io/badger/v3/options"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"sync"
+)
+
+// EntryStore implements the Storage interface backed by
+// an SS table. Specifically we use the implementation
+// from Badger for storing entries in sorted manner.
+type EntryStore struct {
+	sync.Mutex
+	io.Closer
+	db *badger.DB
+	hardState pb.HardState
+	snapshot  pb.Snapshot
+}
+
+func NewTempEntryStore(nodeId uint64) (*EntryStore, error) {
+	if entDir, err := ioutil.TempDir(os.TempDir(), fmt.Sprintf("entries_%x", nodeId)); err != nil {
+		return nil, err
+	} else {
+		log.Printf("[Node %x] Creating RAFT entry store inside the folder: %s", nodeId, entDir)
+		return NewEntryStore(entDir)
+	}
+}
+
+func NewEntryStore(entryDir string) (*EntryStore, error) {
+	options := badger.DefaultOptions(entryDir).
+		WithCompression(bdgr_opts.None).
+		WithSyncWrites(true).
+		WithMetricsEnabled(false).
+		WithLoggingLevel(badger.ERROR).
+		WithDetectConflicts(false)
+	db, err := badger.Open(options)
+	if err != nil {
+		return nil, err
+	}
+	ent := &EntryStore{db: db}
+	err = ent.appendEntries(make([]pb.Entry, 1))
```

Comment:
Does this add anything? Or is this to just assert?

Reply:
Yes this is mainly to maintain parity with the existing impl. An earlier version of my impl. did not have this initial entry in Badger and this resulted in several errors at runtime. Hence retaining it as is.